oml/datasets/audios.py

```diff
@@ -0,0 +1,353 @@
+            extra_data (Optional[Dict[str, Any]]): Extra data to include in dataset items.
+            sr (int): Sampling rate of audio files.
+            num_seconds (float): Duration to use from each audio file.
+            num_channels (int): Number of audio channels.
+            input_tensors_key (str): Key under which audio tensors are stored.
+            input_frame_offset_key (str): Key under which audio offsets are stored
+            index_key (str): Key for indexing dataset items.
+        """
+        self.sr = sr
+        self.num_frames = int(num_seconds * sr)
+        self.num_channels = num_channels
+
+        if extra_data is not None:
+            assert all(
+                len(record) == len(paths) for record in extra_data.values()
+            ), "All the extra records need to have the size equal to the dataset's size"
+            self.extra_data = extra_data
+        else:
+            self.extra_data = {}
+
+        self.frame_offsets: List[int] = self.extra_data.get(input_frame_offset_key)
+
+        self.input_tensors_key = input_tensors_key
+        self.index_key = index_key
+
+        self._paths = [Path(p) if dataset_root is None else Path(dataset_root) / p for p in paths]
+
+    def _downmix_and_resample(self, audio: FloatTensor, sr: int) -> FloatTensor:
+        """
+        Downmix audio to mono and resample it to the dataset's sampling rate.
+
+        Args:
+            audio (FloatTensor): Input audio tensor.
+            sr (int): Original sampling rate of the audio.
+
+        Returns:
+            FloatTensor: Processed audio tensor.
+        """
+        if audio.shape[0] != self.num_channels:
+            audio = audio.mean(dim=1, keepdim=True)
+        if sr != self.sr:
+            resampler = Resample(sr, self.sr)
+            audio = resampler(audio)
+        return audio
+
+    def _trim_or_pad(self, audio: FloatTensor, frame_offset: int) -> FloatTensor:
+        """
+        Trim or pad the audio to match the desired number of frames.
+
+        Args:
+            audio (FloatTensor): Audio tensor.
+            frame_offset (int): Starting frame offset for trimming.
+
+        Returns:
+            FloatTensor: Trimmed or padded audio tensor.
+        """
+        audio_length = audio.shape[1]
+        if audio_length > self.num_frames:
+            if frame_offset is None:
+                frame_offset = random.randrange(0, audio_length - self.num_frames)
+            audio = audio[:, frame_offset : frame_offset + self.num_frames]
+        else:
+            padding = (self.num_frames - audio_length, 0)
+            audio = torch.nn.functional.pad(audio, padding)
+        return audio
+
+    def get_audio(self, item: int) -> FloatTensor:
+        """
+        Load and process an audio file.
+
+        Args:
+            item (int): Dataset item index.
+
+        Returns:
+            FloatTensor: Processed audio tensor.
+        """
+        path = self._paths[item]
+        assert (
```

Comment:
probably it's too late to check an extension, could we fail earlier? in init?

Reply:
done